src/components/form/__tests__/FormField-test.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { screen } from '@testing-library/react';
+import { render } from '~common/helpers/test-utils';
+import { FormField, FormFieldValidation, FormFieldWidth } from '../FormField';
+
+it('displays a label', () => {
+  render(<FormField label="Label 1" />);
+  const label = screen.getByText('Label 1');
+  expect(label).toBeVisible();
+  expect(label.tagName).toBe('LABEL');
+});
+
+it('displays an asterisk next to the label if the form field is required', () => {
+  render(<FormField isRequired label="Label 2" />);
+
+  const label = screen.getByText((_, element) => element?.textContent === 'Label 2*', {
+    selector: 'label',
+  });
+
+  expect(label).toBeVisible();
```

Comment:
A simple `screen.getByLabelText('Input Label*');` doesn't work?

Reply:
Unfortunately it does not. By default testing-library doesn't match text spanning multiple elements, and the asterisks is wrapped in a span tag.

Reply:
Or does it not work because we are not passing any children? So there is actually nothing being labeled?
I'm saying this because I'm using `screen.getByLabelText('Input Label*')` in a TextInput test and it's able to match it somehow.

Reply:
Ah yes, you're right. When it is associated with a labelable element it works!